### PR TITLE
Refresh plugin for June 2023

### PR DESCRIPTION
**Pull request by basil**

### CODE REVIEW

File: .mvn/extensions.xml [hunk 1]
```diff
@@ -2,6 +2,6 @@
   <extension>
     <groupId>io.jenkins.tools.incrementals</groupId>
     <artifactId>git-changelist-maven-extension</artifactId>
-    <version>1.4</version>
+    <version>1.6</version>
```
Comment (basil) [inline .mvn/extensions.xml:5]: Latest at the time of this writing.

File: Jenkinsfile [hunk 2]
```diff
@@ -1,2 +1,5 @@
 // Build the plugin using https://github.com/jenkins-infra/pipeline-library
-buildPlugin()
+buildPlugin(useContainerAgent: true, configurations: [
+  [platform: 'linux', jdk: 17],
+  [platform: 'windows', jdk: 11],
+])
```
Comment (basil) [inline Jenkinsfile:5]: Following the latest archetype.

File: pom.xml [hunk 3]
```diff
@@ -28,27 +28,28 @@
...
-    <gitHubRepo>jenkinsci/publish-over-ssh-plugin</gitHubRepo>
-    <jenkins.version>2.289.3</jenkins.version>
-    <java.level>8</java.level>
-  </properties>
+    <properties>
+        <revision>1.25</revision>
+        <changelist>-SNAPSHOT</changelist>
+        <gitHubRepo>jenkinsci/${project.artifactId}-plugin</gitHubRepo>
+        <jenkins.version>2.361.4</jenkins.version>
+    </properties>
```
Comment (basil) [inline pom.xml:40]: Fixing indentation and making this consistent with the archetype.

File: pom.xml [hunk 4]
```diff
@@ -28,27 +28,28 @@
...
 
   <artifactId>publish-over-ssh</artifactId>
     <packaging>hpi</packaging>
     <name>Publish Over SSH</name>
     <version>${revision}${changelist}</version>
     <description>Send build artifacts over SSH</description>
-    <url>https://github.com/jenkinsci/publish-over-ssh-plugin</url>
+    <url>https://github.com/jenkinsci/${project.artifactId}-plugin</url>
```
Comment (basil) [inline pom.xml:47]: Consistent with the archetype.

File: pom.xml [hunk 5]
```diff
@@ -28,27 +28,28 @@
...
 
     <licenses>
         <license>
             <name>The MIT license</name>
+            <url>https://opensource.org/license/mit/</url>
```
Comment (basil) [inline pom.xml:52]: Consistent with the archetype.

File: pom.xml [hunk 6]
```diff
@@ -66,8 +67,8 @@
         <dependencies>
             <dependency>
                 <groupId>io.jenkins.tools.bom</groupId>
-                <artifactId>bom-2.289.x</artifactId>
-                <version>1500.ve4d05cd32975</version>
+                <artifactId>bom-2.361.x</artifactId>
+                <version>2102.v854b_fec19c92</version>
```
Comment (basil) [inline pom.xml:71]: Latest release of the 2.361.x BOM line.

File: pom.xml [hunk 7]
```diff
@@ -88,11 +89,10 @@
...
         </dependency>
-      <dependency>
-        <groupId>com.fasterxml.jackson.core</groupId>
-        <artifactId>jackson-databind</artifactId>
-        <version>2.14.0</version>
-      </dependency>
+        <dependency>
+            <groupId>org.jenkins-ci.plugins</groupId>
+            <artifactId>jackson2-api</artifactId>
+        </dependency>
```
Comment (basil) [inline pom.xml:95]: Using a library plugin to implement dynamic linking as recommended when such library plugins exist.

File: src/test/java/jenkins/plugins/publish_over_ssh/jenkins/IntegrationTest.java [hunk 8]
```diff
@@ -24,9 +24,9 @@
 
 package jenkins.plugins.publish_over_ssh.jenkins;
 
-import static org.mockito.Matchers.anyObject;
-import static org.mockito.Matchers.anyString;
-import static org.mockito.Matchers.eq;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.ArgumentMatchers.eq;
```
Comment (basil) [inline src/test/java/jenkins/plugins/publish_over_ssh/jenkins/IntegrationTest.java:29]: Adapting to recent versions of Mockito.

**Review by gmcdonald — APPROVED** [review]
Thanks @basil this is great. I have tested the PR locally and works well.
Will merge , thanks again

**Pull request merged**